packages/ng/form-field/form-field.component.ts

```diff
@@ -30,130 +31,87 @@ let nextId = 0;
+		return hasRequiredNgModel || hasRequiredFormControl;
+	});
 
-	@Input({
-		required: true,
-	})
-	label: PortalContent;
+	label = input.required<PortalContent>();
 
 	/**
 	 * Hide field label, while keeping it in DOM for screen readers
 	 */
-	@Input({
-		transform: booleanAttribute,
-	})
-	hiddenLabel = false;
+	hiddenLabel = input<boolean, boolean>(false, { transform: booleanAttribute });
 
-	@Input({
-		transform: booleanAttribute,
-	})
-	rolePresentationLabel = false;
+	rolePresentationLabel = input<boolean, boolean>(false, { transform: booleanAttribute });
 
-	@Input({
-		transform: booleanAttribute,
-	})
-	inline = false;
+	inline = input<boolean, boolean>(false, { transform: booleanAttribute });
 
-	@Input()
-	statusControl: AbstractControl;
+	statusControl = input<AbstractControl | null>(null);
 
-	@Input()
-	tooltip: string | SafeHtml;
+	tooltip = input<string | SafeHtml | null>(null);
 
-	/**
-	 * Override from input
-	 * @private
-	 */
-	#invalidStatusOverride = false;
-
-	@Input({ transform: booleanAttribute })
-	set invalid(invalid: boolean) {
-		this.#invalidStatusOverride = invalid !== undefined && invalid !== null;
-		this.invalidStatus = invalid;
-		this.updateAria();
-	}
+	invalidStatus = computed(() => {
+		const isInvalidOverride = this.invalid() !== undefined && this.invalid() !== null;
+		if (isInvalidOverride) {
+			return isInvalidOverride;
+		}
+		const statusControlOverride = this.statusControl();
+		if (statusControlOverride) {
+			return statusControlOverride.invalid && this.refreshedOwnControls().some((c) => c.touched);
+		}
+		return this.refreshedOwnControls().some((c) => c.invalid && c.touched);
+	});
 
-	/**
-	 * Used to cache previous invalid status and know if we want to update aria stuff or not.
-	 * @private
-	 */
-	protected invalidStatus = false;
+	invalid = input<boolean | null, boolean>(null, { transform: booleanAttribute });
 
-	@Input()
-	inlineMessage: PortalContent;
+	inlineMessage = input<PortalContent | null>(null);
 
 	/**
 	 * Inline message for when the control is in error state
 	 */
-	@Input()
-	errorInlineMessage: PortalContent;
+	errorInlineMessage = input<PortalContent | null>(null);
 
 	/**
 	 * State of the inline message, will be ignored if form state is invalid
 	 */
-	@Input()
-	inlineMessageState: InlineMessageState;
+	inlineMessageState = input<InlineMessageState | null>(null);
 
-	@Input()
-	size: FormFieldSize;
+	size = input<FormFieldSize | null>(null);
 
-	@Input()
-	layout: 'default' | 'checkable' | 'fieldset' = 'default';
+	layout = model<'default' | 'checkable' | 'fieldset'>('default');
 
 	hasArrow = false;
 
 	#inputs: InputDirective[] = [];
+
 	/**
 	 * Max amount of characters allowed, defaults to 0, which means hidden, no maximum
 	 */
-	@Input()
-	counter = 0;
-
-	required = false;
+	counter = input<number>(0);
 
 	get contentLength(): number {
 		return (this.#inputs[0]?.host?.nativeElement as HTMLInputElement)?.value.length || 0;
```

Comment:
If `#inputs` becomes a `Signal`, we could have a `computed` for `contentLength` too

Reply:
I don't think we can make it a signal due to how it's built and updated

Reply:
Indeed, we would have to listen to input/change event